assets/react/v3/shared/molecules/DropdownButton.tsx

```diff
@@ -0,0 +1,442 @@
+import React, { MouseEvent, ReactNode, useRef, useState } from 'react';
+import { SerializedStyles, css, keyframes } from '@emotion/react';
+
+import { ButtonIconPosition, ButtonSize, ButtonVariant } from '@Atoms/Button';
+import SVGIcon from '@Atoms/SVGIcon';
+
+import { borderRadius, colorTokens, fontSize, fontWeight, lineHeight, shadow, spacing, zIndex } from '@Config/styles';
+import { styleUtils } from '@Utils/style-utils';
+import { AnimationType } from '@Hooks/useAnimation';
+
+import Popover from './Popover';
+
+interface DropdownOptionProps {
+  text: string | ReactNode;
+  disabled?: boolean;
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  buttonContentCss?: SerializedStyles;
+}
+
+export const DropdownItem = ({ text, disabled = false, onClick, buttonContentCss }: DropdownOptionProps) => {
+  return (
+    <button type="button" css={styles.dropdownOption(disabled)} onClick={onClick}>
+      <span css={[styles.dropdownOptionContent, buttonContentCss]}>{text}</span>
+    </button>
+  );
+};
+
+interface DropdownButtonProps {
+  text: string | ReactNode;
+  children: ReactNode;
+  variant?: ButtonVariant;
+  type?: 'submit' | 'button';
+  size?: ButtonSize;
+  icon?: React.ReactNode;
+  iconPosition?: ButtonIconPosition;
+  disabled?: boolean;
+  loading?: boolean;
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  tabIndex?: number;
+  buttonCss?: SerializedStyles;
+  buttonContentCss?: SerializedStyles;
+  dropdownMaxWidth?: string;
+}
+
+const DropdownButton = ({
+  type = 'button',
+  text,
+  children,
+  variant = 'primary',
+  size = 'medium',
+  icon,
+  iconPosition = 'left',
+  loading = false,
+  disabled = false,
+  tabIndex,
+  onClick,
+  buttonCss,
+  buttonContentCss,
+  dropdownMaxWidth = '140px',
+}: DropdownButtonProps) => {
+  const dropdownTriggerRef = useRef<HTMLButtonElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <>
+      <div css={styles.wrapper}>
+        <button
+          type={type}
+          css={[styles.button(variant, size, loading, disabled), buttonCss]}
+          onClick={onClick}
+          tabIndex={tabIndex}
+        >
+          {loading && !disabled && (
+            <span css={styles.spinner}>
+              <SVGIcon name="spinner" width={18} height={18} />
+            </span>
+          )}
+          <span css={[styles.buttonContent(loading, disabled), buttonContentCss]}>
+            {icon && iconPosition === 'left' && <span css={styles.buttonIcon(iconPosition)}>{icon}</span>}
+            {text}
+            {icon && iconPosition === 'right' && <span css={styles.buttonIcon(iconPosition)}>{icon}</span>}
+          </span>
+        </button>
+        <button
+          ref={dropdownTriggerRef}
+          type={'button'}
+          css={[styles.button(variant, size, (loading = false), disabled), styles.dropdownButton(variant, disabled)]}
```

Comment:
Could you please explain this `(loading = false)` thing?

Reply:
Both of them use the same style so both can have the same style, but we don't need the dropdown trigger button to have a loading state. We are receiving loading props for the main button. To make sure it does not have any effect from loading props, it is defaulted as false.